packages/atclient/src/atkeys.c

```diff
@@ -622,57 +672,90 @@ int atclient_atkeys_populate_from_strings(atclient_atkeys *atkeys, const char *a
-                                               strlen(atkeys->encrypt_public_key_base64))) != 0) {
+                                                 strlen(atkeys->encrypt_public_key_base64))) != 0) {
     atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR,
                  "atchops_rsa_key_populate_public_key: %d | failed to populate encrypt public key\n", ret);
     goto exit;
   }
 
+  // 6. enrollment id, if it exists
+  if (enrollment_id_str != NULL && enrollment_id_str_len > 0) {
+    if ((ret = atclient_atkeys_set_enrollment_id(atkeys, enrollment_id_str, enrollment_id_str_len)) != 0) {
+      atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "atclient_atkeys_set_enrollment_id: %d\n", ret);
+      goto exit;
+    }
+  }
+
   ret = 0;
   goto exit;
 
 exit: { return ret; }
 }
 
-int atclient_atkeys_populate_from_atkeysfile(atclient_atkeys *atkeys, const atclient_atkeysfile atkeysfile) {
+int atclient_atkeys_populate_from_atkeys_file(atclient_atkeys *atkeys, const atclient_atkeysfile *atkeys_file) {
   int ret = 1;
 
-  ret = atclient_atkeys_populate_from_strings(
-      atkeys, atkeysfile.aes_pkam_public_key_str, strlen(atkeysfile.aes_pkam_public_key_str), atkeysfile.aes_pkam_private_key_str,
-      strlen(atkeysfile.aes_pkam_private_key_str), atkeysfile.aes_encrypt_public_key_str,
-      strlen(atkeysfile.aes_encrypt_public_key_str), atkeysfile.aes_encrypt_private_key_str,
-      strlen(atkeysfile.aes_encrypt_private_key_str), atkeysfile.self_encryption_key_str,
-      strlen(atkeysfile.self_encryption_key_str));
-  if (ret != 0) {
-    atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR,
-                 "atclient_atkeys_populate_from_strings: %d | failed to populate from strings\n", ret);
+  if (atkeys == NULL) {
+    ret = 1;
+    atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "atkeys is NULL\n");
     goto exit;
   }
 
+  if (atkeys_file == NULL) {
+    ret = 1;
+    atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "atkeys_file is NULL\n");
+    goto exit;
+  }
+
+  if (atclient_atkeysfile_is_enrollment_id_str_initialized((atclient_atkeysfile *)atkeys_file)) {
+    if ((ret = atclient_atkeys_populate_from_strings(
+             atkeys, atkeys_file->aes_pkam_public_key_str, strlen(atkeys_file->aes_pkam_public_key_str),
+             atkeys_file->aes_pkam_private_key_str, strlen(atkeys_file->aes_pkam_private_key_str),
+             atkeys_file->aes_encrypt_public_key_str, strlen(atkeys_file->aes_encrypt_public_key_str),
+             atkeys_file->aes_encrypt_private_key_str, strlen(atkeys_file->aes_encrypt_private_key_str),
+             atkeys_file->self_encryption_key_str, strlen(atkeys_file->self_encryption_key_str),
+             atkeys_file->enrollment_id_str, strlen(atkeys_file->enrollment_id_str))) != 0) {
+      atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR,
+                   "atclient_atkeys_populate_from_strings: %d | failed to populate from strings\n", ret);
+      goto exit;
+    }
```

Comment:
conditionally add the enrollment id if it is initialized when populating from atkeys file